pkg/integrations/v4/dm/id_provider.go

```diff
@@ -0,0 +1,102 @@
+package dm
+
+import (
+	"fmt"
+	"github.com/newrelic/infrastructure-agent/pkg/backend/identityapi"
+	"github.com/newrelic/infrastructure-agent/pkg/entity"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/protocol"
+)
+
+type RegisteredEntitiesNameToID map[string]entity.ID
```

Comment:
```suggestion
type registeredEntitiesNameToID map[string]entity.ID
```
`idProviderInterface` isn't exposed so no need to expose these types.

Reply:
we will move it into different package in the next PR - https://github.com/newrelic/infrastructure-agent/pull/76/files?file-filters%5B%5D=#diff-7bb3595cb8057cbf50904c707b314e31R147